Libraries/acdriver/Headers/acdriver/Extensions.h

```diff
@@ -0,0 +1,84 @@
+/**
+ Copyright (c) 2015-present, Facebook, Inc.
+ All rights reserved.
+
+ This source code is licensed under the BSD-style license found in the
+ LICENSE file in the root directory of this source tree. An additional grant
+ of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#ifndef __acdriver_Extensions_h
+#define __acdriver_Extensions_h
+
+#include <acdriver/Result.h>
+#include <car/Rendition.h>
+
+#include <algorithm>
+#include <ext/optional>
+#include <string>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+
+namespace acdriver {
+
+/*
+ * Contains non-standard related logic
+ */
+class Extensions {
+public:
+    /*
+     *  Extra image types (besides PNG and JPEG) to include in ImageSet
+     */
+    enum class ImageType {
+        WEBP,
+    };
+
+    class ImageTypeHash
+    {
+    public:
+        std::size_t operator()(ImageType t) const
+        {
+            return static_cast<std::size_t>(t);
+        }
+    };
+
+    typedef std::unordered_set<ImageType, ImageTypeHash> ImageTypeSet;
```

Comment:
Why is this an std::unoreder_set? That seems pretty heavy-weight. Why not a plain C struct containing boolean members?

```cpp
struct ImageTypeSet {
    bool webp = false;
};
```

Reply:
I agree it's heavy... but I still prefer unordered_set with that enum key just to make it easier to add new file types...
That structure is not going to be used that much...
I can still change it, and have member functions to translate the enums if you like it...